.eslintrc.js

```diff
@@ -36,6 +36,15 @@ module.exports = {
       rules: {
         '@typescript-eslint/no-explicit-any': 'error',
         '@typescript-eslint/no-shadow': ['error', { allow: ['Text'] }],
+        '@typescript-eslint/consistent-type-imports': [
+          'error',
+          {
+            prefer: 'type-imports',
+            disallowTypeAnnotations: true,
+            fixStyle: 'separate-type-imports',
+          },
+        ],
+        '@typescript-eslint/no-import-type-side-effects': 'error',
```

Comment:
This ESLint rule enforces using `import type` syntax for type imports, which helps prevent runtime errors in `.mjs` files where type imports aren't supported. It ensures type imports are properly separated from value imports, making the code more maintainable and compatible with ECMAScript modules.

### Before
<img width="724" alt="Screenshot 2024-12-02 at 4 40 41 PM" src="https://github.com/user-attachments/assets/db7b4a26-27fa-4360-acc1-79f2671e1cb7">

### After
<img width="662" alt="Screenshot 2024-12-02 at 4 43 10 PM" src="https://github.com/user-attachments/assets/36555084-4514-45f3-9d1b-e6f5b2b64ca3">

Reply:
Revert changes already in the MM eslint plugin https://github.com/MetaMask/eslint-config/blob/733b9e2835c5422a2f222b23cf373f4050936409/packages/typescript/src/index.mjs#L48